src/WebJobs.Extensions.DurableTask/DurableHttpResponse.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    /// <summary>
+    /// Response received from the HTTP request made by the Durable Function.
+    /// </summary>
+    public class DurableHttpResponse
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DurableHttpResponse"/> class.
+        /// </summary>
+        /// <param name="statusCode">HTTP Status code returned from the HTTP call.</param>
+        public DurableHttpResponse(HttpStatusCode statusCode)
```

Comment:
I suggest you create another constructor that takes `HttpResponseMessage`. I saw that you have logic elsewhere that manually creates `DurableHttpResponse` objects from `HttpResponseMessage` and then sets its properties. This is not ideal because it requires all the properties to be settable. A cleaner design would be to only allow the `DurableHttpresponse` class to manipulate its own internal state.

Reply:
And it looks like we may also need a constructor for `JObject`. That can replace the logic inside the `DurableHttpResponseJsonConverter` which is directly manipulating `DurableHttpResponse`.